src/Parsers/ExpressionListParsers.cpp

```diff
@@ -786,6 +789,9 @@ template <TokenType separator, TokenType end>
 class BaseLayer : public Layer
```

Comment:
I think it's a bit misleading that BaseLayer extends from Layer. Maybe the names can be interchanged ?
But I see in other Layers, that we sometimes extend from Layer and sometimes from BaseLayer, not sure what names would be helpful.

Reply:
We have classes `IParser` and `IParserBase` where `IParserBase` extends from `IParser`, so it was done here in a similar way. We can maybe change `BaseLayer` to `OrdinaryLayer` or something similar to that, what do you think?

Reply:
I was not aware of similar patterns, thanks. But I see that almost all Parsers use IParserBase.
In our case we extend from both, I think the difference is the separator & end. How about LayerWithSeparator  or LayerWithCustomSeparator? I am not a big fan of OrdinaryLayer, but I checked in our code we use OrdinaryXXX / SimpleXXX names, so to be consistent these would do too. Let me know your thoughts

Reply:
I think for now LayerWithSeparator is ok for now. I just want to deliver this alias fix before the 22.10 release.